export.py

```diff
@@ -1779,6 +1779,21 @@ def has_emissive_material(db):
     return False
 
 
+def export_for_bake(db):
+    export = True
+    node_names = []
+    for mat in db.material:
+        if mat and mat.node_tree:
+            nt = mat.node_tree
+            for n in nt.nodes:
+                node_names.append(n.name)
+    if node_names:
+        if 'PxrBakeTexture' not in node_names:
+            export = False
+
```

Comment:
Hmmm... so the only instance you have to return true and end is like you said when you find a bake texture node (by bl_idname).  Otherwise once you get to the end of the loop after all materials return false

Cameras don't worry about.  I think those can be skipped.  It should get the right one anyway in the export_camera method.